### PR TITLE
chore: added PR template

### PR DESCRIPTION
**Pull request by BartoszJarocki**
This PR introduces a PR template. The main reason for listing technically the obvious things is forming the habit of actually sharing the proper information with the hope it'll speed up the review process. 

let's discuss!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Fixes #[issue number]
+[Please include a summary of the changes and the related issue]
+
+## Demo
+
+[If possible, please include a screenshot or gif/video, it'll make it easier for reviewers to understand the scope of the changes and how the change is supposed to work. If you're introducing something new or changing the existing patterns, please share a Loom and explain what decisions you've made and under what circumstances]
+
+## Release notes
+
+[Include a human-readable description that could help to prepare a changelog]
```
Comment (Dschoordsch) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:12]: -1 I think the PR title should be that human readable line.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Description
+
+Fixes #[issue number]
+[Please include a summary of the changes and the related issue]
+
+## Demo
+
+[If possible, please include a screenshot or gif/video, it'll make it easier for reviewers to understand the scope of the changes and how the change is supposed to work. If you're introducing something new or changing the existing patterns, please share a Loom and explain what decisions you've made and under what circumstances]
+
+## Release notes
```
Comment (igorlesnenko) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:10]: Should we parse that section in future to automatically generate PR desctiption for changelog?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- [ ] Scenario B
+  - Step 1
+  - Step 2....
+
+## Final checklist
+
+- [ ] I checked the [code review guidelines](../../docs/codeReview.md)
+- [ ] I have performed a self-review of my code, the same way I'd do it for any other team member
+- [ ] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
+- [ ] I have commented on my code, particularly in hard-to-understand areas
```
Comment (Dschoordsch) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:31]: +1 I have some reservations listing it like this, maybe change it to
> I added a comment when I took a non-obvious choice why I did it this way

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 2 above)
Comment (BartoszJarocki) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:10]: yes, I'd like us to do it at some point

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- [ ] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns
```
Comment (Dschoordsch) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:32]: +1 I would add "and I think one review is sufficient" to let people know they still have a choice here.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (BartoszJarocki) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:12]: good point

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 3 above)
Comment (BartoszJarocki) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:31]: what about 
> Whenever I took a non-obvious choice I added a comment explaining why I did it this way 
?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 4 above)
Comment (BartoszJarocki) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:32]: perfect

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (BartoszJarocki) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: added a checklist item, @Dschoordsch I would appreciate it if you could take a look

**Review by nickoferrall — APPROVED** [review]

**Review by tianrunhe — CHANGES_REQUESTED** [review]

**Review by Dschoordsch — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+- [ ] Scenario A
+  - Step 1
+  - Step 2...
+
+- [ ] Scenario B
+  - Step 1
+  - Step 2....
+
+## Final checklist
```
[the remark this answers is not in the document]
Comment (BartoszJarocki) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:22]: @tianrunhe I think you only get the workflows to run _after_ submitting a PR. failed workflow is basically a no-go anyway so I don't see too much value in including that in the checklist

[the remark this answers is not in the document]
Comment (tianrunhe) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:22]: Oh yeah that's correct!

**Review by tianrunhe — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+- [ ] I checked the [code review guidelines](../../docs/codeReview.md)
+- [ ] I have performed a self-review of my code, the same way I'd do it for any other team member
+- [ ] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
+- [ ] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
+- [ ] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
+- [ ] PR title is human readable and could be used in changelog
```
Comment (mattkrick) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:29]: explain that it is a conventional commit & link to the docs?

**Pull request merged**